mast_aladin_lite/aida.py

```diff
@@ -34,12 +38,35 @@ def set_viewport(self, center):
 
         """
 
-        if not isinstance(center, SkyCoord):
-            raise TypeError(
-                "`center` must be a SkyCoord object."
-            )
+        if center is not None:
+            if not isinstance(center, SkyCoord):
+                raise TypeError(
+                    "`center` must be a SkyCoord object."
+                )
+
+            self.app.target = center
+
+        if fov is not None:
+            scale_factor = 0
+            if isinstance(fov, (u.Quantity, Angle)):
+                fov = fov.value
+
+            elif not isinstance(fov, (float, int)):
+                raise ValueError(
+                    f"`fov` must be an `~astropy.coordinates.Angle` or float, got {fov=}"
+                )
 
-        self.app.target = center
+            current_fov = self.app.fov.to_value(u.deg)
+            aspect_ratio = float(self.app._fov_xy["y"] / self.app._fov_xy["x"])
```

Comment:
I had to look this up since different languages do different things, but I think these `float()` calls are redundant because Python division always returns a float.    
https://docs.python.org/3/library/operator.html#mapping-operators-to-functions
https://docs.python.org/3/library/operator.html#mapping-operators-to-functions

I was checking because if the division could have returned a rounded-off int when dividing 2 ints, then this `float()` should be on one (or both) of the operands instead of on the result which would already have been rounded.